### PR TITLE
Celery requeue tasks

### PR DESCRIPTION
**Pull request by nooraangelva**
Ref: #2442

### CODE REVIEW

File: backend/inspirehep/celery.py [hunk 1]
```diff
@@ -34,6 +35,8 @@ def on_failure(self, exc, task_id, args, kwargs, einfo):
                     term_log.write(str(exc))
             finally:
                 os.kill(os.getppid(), signal.SIGTERM)
+        else:
+            Reject(requeue=True)
```
Comment (drjova) [inline backend/inspirehep/celery.py:39]: Could you please add a test for this?

Comment (nooraangelva) [reply backend/inspirehep/celery.py:39]: Sure. Are there any tests I could use as a reference? I haven't done any tests for celery yet, so I'm not sure where to create one.

Comment (michamos) [reply backend/inspirehep/celery.py:39]: wouldn't you need to `raise` this, as it's an exception? :thinking:

Comment (nooraangelva) [reply backend/inspirehep/celery.py:39]: That is my understanding also from the documentation, but when I raise it I get this error from several tests ```celery.exceptions.Reject: (AttributeError("'NoneType' object has no attribute 'create_ticket'"), True)```

**Comment by nooraangelva** [comment]
> @nooraangelva when everything is passing could you please rename the commit message? It has a typo 🙃

sure thing 👍

File: backend/tests/integration/submissions/test_submissions_views.py [hunk 2]
```diff
@@ -1727,8 +1733,9 @@ def test_confirmation_email_not_sent_when_user_is_cataloger(
 
 
 @patch("inspirehep.submissions.views.send_conference_confirmation_email")
+@patch("inspirehep.submissions.views.async_create_ticket_with_template")
```
Comment (drjova) [inline backend/tests/integration/submissions/test_submissions_views.py:1736]: You don't need this `patch` it has already :)

Comment (nooraangelva) [reply backend/tests/integration/submissions/test_submissions_views.py]: hmm, okay but it then has the same error as the ones that didn't have the ```patch```

Comment (drjova) [reply backend/tests/integration/submissions/test_submissions_views.py]: Ok I see, then we have to put this patch as well.

Comment (nooraangelva) [reply backend/tests/integration/submissions/test_submissions_views.py]: yep, then just need to figure out a way to 'fix' the ```assertion error``` it creates.

File: backend/tests/unit/submissions/test_tasks.py [hunk 3]
```diff
@@ -31,3 +33,23 @@ def test_async_create_ticket_with_template_logs_on_error(mock_create_ticket):
...
+        {"foo": "bar"},
+        "Test Ticket",
+        12345,
+    ]
+
+    mock_create_ticket.side_effect = InvalidRequestError
+    with pytest.raises(InvalidRequestError):
+        async_create_ticket_with_template(*args)
+
+    mock_create_ticket.side_effect = Reject()
```
Comment (drjova) [inline backend/tests/unit/submissions/test_tasks.py:53]: I would change that to any `Exception` and it should raise the `Reject`

**Comment by michamos** [comment]
I have a stupid question: won't tasks get requeued forever? What about cases where the task actually can't succeed due to various reasons (as happens often during ORCID push or HAL push). Wouldn't that be a problem?

**Comment by nooraangelva** [comment]
That's actually a good point. there is a warning in the documentation about the infinite loop.

I was thinking, that instead of ```Reject()```  we could use ```Retry()``` instead. That can have ```max_retries``` so there is no infinite loop.

**Comment by nooraangelva** [comment]
So what would be the best solution for this? Any thoughts @drjova?

**Comment by drjova** [comment]
We are already retrying the tasks, I think we should keep doing that instead of a global solution.

**Comment by nooraangelva** [comment]
Okay, so this PR and issue can be closed then?

**Pull request closed without merging**